docs/cli.md

```diff
@@ -147,31 +147,41 @@ For instance, for ES6 support with [Babel](https://babeljs.io/) 7 add:
 
 This will effectivally call `require('@babel/register')` prior to requiring any support files.
 
-### Non JS files
+### Typescript (and other non-JS files)
 
-If your files end in an extension other than `js`, make sure to also include the `--require` option to state the support files to require.
+If your files end with an extension other than `js`, make sure to also include the `--require` option to state the required support files.
 
-For example, with [TypeScript](https://www.typescriptlang.org/):
+When using babel with [@babel/preset-typescript](https://babeljs.io/docs/en/babel-preset-typescript):
+
+```
+--require-module @babel/register --require 'step-definitions/**/*.ts'
+```
+
+If not using babel you can use [ts-node](https://github.com/TypeStrong/ts-node):
 
 ```
 --require-module ts-node/register --require 'step-definitions/**/*.ts'
 ```
 
-or [CoffeeScript](https://www.npmjs.com/package/coffeescript):
+> ⚠️ Note the Extra Configuration section below for more required settings for this setup
+
+or for [CoffeeScript](https://www.npmjs.com/package/coffeescript):
```

Comment:
Lets just delete coffeescript and then the Note section could probably be deleted

Reply:
I kept the CoffeeScript example in order to illustrate the transpilation section before the dedicated chapter on typescript